documentation/sphinx/source/class-scheduling-go.rst

```diff
@@ -169,6 +173,7 @@ We're going to rely on the powerful guarantees of transactions to help keep all
 
   func signup(t fdb.Transactor, studentID, class string) (err error) {
     _, err = t.Transact(func (tr fdb.Transaction) (ret interface{}, err error) {
+      tr.Options().SetTimeout(3000)
```

Comment:
I feel kind of uneasy about this change. If the provided `fdb.Transactor` is a `fdb.Database`, then I think this is fine. But if this is an `fdb.Transaction`, then I think this will either override an existing timeout...which I don't think is correct. It certainly wouldn't be the behavior I would expect if I were a client.

Reply:
That's valid. There's not enough code here as it should be a `fdb.Database`. I added a complete executable code example which hopefully will make things more clear for the reader.

Reply:
I think it's "correct" as a `Transactor`, it's just that the option only needs to be set if it's an `fdb.Database`. Something like:

```go
switch transactorType := t.(type) {
case fdb.Database:
   tr.Options().SetTimeout(3000)
default:
}
```

I'm not actually sure if that compiles.